client/core/bookie.go

```diff
@@ -651,6 +652,18 @@ func (dc *dexConnection) refreshServerConfig() error {
 	dc.assets = assets
 	dc.assetsMtx.Unlock()
 
+	// If we're fetching config and the server sends the pubkey in config, set
+	// the dexPubKey now. We also know that we're fetching the config for the
+	// first time (via connectDEX), and the dexConnection has not been assigned
+	// to dc.conns yet, so we can still update the acct.dexPubKey field without
+	// a data race.
+	if dc.acct.dexPubKey == nil && len(cfg.DEXPubKey) > 0 {
+		dc.acct.dexPubKey, err = secp256k1.ParsePubKey(cfg.DEXPubKey)
+		if err != nil {
```

Comment:
I started making a similar change in the postbond branch because I needed DEXPubKey in the config response too, but `refreshServerConfig` is called on each reconnect.  So I had `connectDEX` do this with a `cfg` returned from `refreshServerConfig`: https://github.com/decred/dcrdex/pull/1120/commits/7c98429574c2f7826dd9b95af8777ba9113c639f#diff-13a4a6b12c1ae844ef90151638f194427cfdc545fceb58d52c0f6f33b81d859bR4940-R4953
And `handleReconnect` (the other caller of `refreshServerConfig`) doesn't touch `cfg.DEXPubKey`.
Is the idea here that `dc.acct.dexPubKey == nil` signifies whet it is actually the first connect? (*if* we're fetching the config for the first time)? I suppose that does make this safe with `cfgMtx` held...
I can do the same thing if that's the intent.

Reply:
> Is the idea here that dc.acct.dexPubKey == nil signifies when it is actually the first connect? 

That is the idea, but please to improve things in postbond if you can. It's a little roundabout.